src/query_plan/fetch_dependency_graph.rs

```diff
@@ -192,25 +192,18 @@ pub(crate) struct DeferTracking {
 #[derive(Debug, Clone)]
 pub(crate) struct DeferredInfo {
     pub(crate) label: DeferRef,
-    pub(crate) path: FetchDependencyGraphPath,
+    pub(crate) path: FetchDependencyGraphNodePath,
     pub(crate) sub_selection: NormalizedSelectionSet,
     pub(crate) deferred: IndexSet<DeferRef>,
     pub(crate) dependencies: IndexSet<DeferRef>,
 }
 
 // TODO: Write docstrings
-#[derive(Debug, Clone)]
-pub(crate) struct FetchDependencyGraphPath {
-    pub(crate) full_path: OpPath,
-    pub(crate) path_in_node: OpPath,
-    pub(crate) response_path: Vec<FetchDataPathElement>,
-}
-
 #[derive(Debug, Clone, Default)]
 pub(crate) struct FetchDependencyGraphNodePath {
-    full_path: Arc<OpPath>,
-    path_in_node: Arc<OpPath>,
-    response_path: Vec<FetchDataPathElement>,
+    pub(crate) full_path: Arc<OpPath>,
+    pub(crate) path_in_node: Arc<OpPath>,
+    pub(crate) response_path: Vec<FetchDataPathElement>,
```

Comment:
Let's limit visibility for fields that don't need to be made public.
```suggestion
    path_in_node: Arc<OpPath>,
    response_path: Vec<FetchDataPathElement>,
```